### PR TITLE
fix: ensure valid output for import the same cjs twice

### PR DESCRIPTION
**Pull request by hyf0**
### Description

https://hyrious.me/esbuild-repl/?version=0.18.19&b=e%00entry.js%00import+%7B+a+%7D+from+%27.%2Ffile%27%0Aconsole.log%28a%29%0Aimport+%7B+a+as+a2+%7D+from+%27.%2Ffile%27%0Aconsole.log%28a2%29&b=%00file.js%00module.exports.a+%3D+1&o=--bundle+--outdir%3D%22%2F%22+--format%3Desm+--splitting

---

This PR also remove concept `ReExport`. A `ReExport` is a combination of `LocalExport` and `NamedExport`. This reduces complexity of dealing with `ReExport`.

### Test Plan

---

### CODE REVIEW

**Comment by hyf0** [comment]
Current dependencies on/for this PR:
* main
  * **PR #315** <a href="https://app.graphite.dev/github/pr/rolldown-rs/rolldown/315?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #316** <a href="https://app.graphite.dev/github/pr/rolldown-rs/rolldown/316?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
      * **PR #317** <a href="https://app.graphite.dev/github/pr/rolldown-rs/rolldown/317?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
        * **PR #319** <a href="https://app.graphite.dev/github/pr/rolldown-rs/rolldown/319?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
          * **PR #320** <a href="https://app.graphite.dev/github/pr/rolldown-rs/rolldown/320?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
            * **PR #321** <a href="https://app.graphite.dev/github/pr/rolldown-rs/rolldown/321?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
              * **PR #322** <a href="https://app.graphite.dev/github/pr/rolldown-rs/rolldown/322?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                * **PR #323** <a href="https://app.graphite.dev/github/pr/rolldown-rs/rolldown/323?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈
                  * **PR #324** <a href="https://app.graphite.dev/github/pr/rolldown-rs/rolldown/324?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                    * **PR #325** <a href="https://app.graphite.dev/github/pr/rolldown-rs/rolldown/325?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                      * **PR #326** <a href="https://app.graphite.dev/github/pr/rolldown-rs/rolldown/326?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                        * **PR #327** <a href="https://app.graphite.dev/github/pr/rolldown-rs/rolldown/327?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                          * **PR #328** <a href="https://app.graphite.dev/github/pr/rolldown-rs/rolldown/328?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                            * **PR #329** <a href="https://app.graphite.dev/github/pr/rolldown-rs/rolldown/329?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                              * **PR #332** <a href="https://app.graphite.dev/github/pr/rolldown-rs/rolldown/332?utm_source=stack-comment-icon" target="_blank">Graphite</a> 

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/rolldown-rs/rolldown/323?utm_source=stack-comment).

File: crates/rolldown/src/bundler/visitors/scanner.rs [hunk 1]
```diff
@@ -124,7 +124,16 @@ impl<'ast> Scanner<'ast> {
   }
 
   fn add_import_record(&mut self, module_request: &Atom, kind: ImportKind) -> ImportRecordId {
-    let rec = RawImportRecord::new(module_request.clone(), kind);
+    let importee_repr = representative_name(module_request);
+    let namespace_ref: SymbolRef = (
+      self.idx,
+      self
+        .symbol_table
+        .create_symbol(format!("import_{importee_repr}").into(), self.scope.root_scope_id()),
```
Comment (underfin) [inline crates/rolldown/src/bundler/visitors/scanner.rs:132]: Here is better if check the symbol is already created, it will reduce one symbol deconfilct.

**Comment by underfin** [comment]
Cool!

**Review by underfin — APPROVED** [review]

File: crates/rolldown/src/bundler/visitors/scanner.rs (hunk 1 above)
Comment (hyf0) [reply crates/rolldown/src/bundler/visitors/scanner.rs:132]: Yeah. I saw that. Currently, we would create symbol for each `ImportReocrd`, this is how esbuild does it too. I'm thinking how to optimize it, but that's another story :).

**Comment by hyf0** [comment]
### Merge activity

* **Nov 19, 7:13 AM**: @hyf0 started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/rolldown-rs/rolldown/323).
* **Nov 19, 7:13 AM**: @hyf0 merged this pull request with [Graphite](https://app.graphite.dev/github/pr/rolldown-rs/rolldown/323).

**Pull request merged**